### PR TITLE
Expose Camera APIs in iOS SDK

### PR DESCRIPTION
**Pull request by msmollin**

### CODE REVIEW

File: ios/src/TGMapViewController.h [hunk 1]
```diff
@@ -22,13 +42,34 @@
...
 @property (assign, nonatomic) BOOL continuous;
-@property (weak, nonatomic) id<TangramGestureRecognizerDelegate> gestureDelegate;
+@property (weak, nonatomic, nullable) id<TangramGestureRecognizerDelegate> gestureDelegate;
+
+// The following are computed property. They return sensible defaults when the above .map property is nil
+@property (assign, nonatomic) TangramCameraType cameraType;
+@property (assign, nonatomic) TangramGeoPoint position;
+@property (assign, nonatomic) float zoom;
+@property (assign, nonatomic) float rotation;
+@property (assign, nonatomic) float cameraTilt;
```
Comment (blair1618) [inline ios/src/TGMapViewController.h]: `tilt` seems more in line with the other property names than `cameraTilt`. The [MapKit camera](https://developer.apple.com/library/ios/documentation/MapKit/Reference/MKMapCamera_class/index.html#//apple_ref/occ/instp/MKMapCamera/pitch) uses "pitch" for this, and "bearing" for rotation (sort of) but we don't have to match it exactly.

File: ios/src/TGMapViewController.h [hunk 2]
```diff
@@ -22,13 +42,34 @@
...
 
 - (void)renderOnce;
 
+- (void)setPositionEasedAtPosition:(TangramGeoPoint)position withDuration:(float)duration;
```
Comment (blair1618) [inline ios/src/TGMapViewController.h]: There's gotta be a better way to name this. Is the idea is to make it read like a sentence? What about `startEasingToPosition: ...` or `animateToPosition: ...`

File: ios/src/TGMapViewController.mm [hunk 3]
```diff
@@ -21,6 +21,131 @@ @interface TGMapViewController ()
...
 @implementation TGMapViewController
 
+- (void)setPosition:(TangramGeoPoint)position {
+    if (self.map) {
+        self.map->setPosition(position.longitude, position.latitude);
+    }
+}
+
+- (void)setPositionEasedAtPosition:(TangramGeoPoint)position withDuration:(float)duration {
+    [self setPositionEasedAtPosition:position withDuration:duration withEaseType:TangramEaseTypeQuint];
```
Comment (blair1618) [inline ios/src/TGMapViewController.mm]: I think we decided to make the default ease type "cubic" for consistency with the Android SDK.

File: ios/src/TGMapViewController.h [hunk 4]
```diff
@@ -22,13 +42,34 @@
...
+
+- (void)setPositionEasedAtPosition:(TangramGeoPoint)position withDuration:(float)duration withEaseType:(TangramEaseType)easeType;
+
+- (void)setZoomEased:(float)zoomLevel withDuration:(float)duration;
+
+- (void)setZoomEased:(float)zoomLevel withDuration:(float)duration withEaseType:(TangramEaseType)easeType;
+
+- (void)setRotationEased:(float)rotation withDuration:(float)duration;
+
+- (void)setRotationEased:(float)rotation withDuration:(float)duration withEaseType:(TangramEaseType)easeType;
```
Comment (blair1618) [inline ios/src/TGMapViewController.h]: Similar to the eased position setter, we could name these along the lines of `startEasingToRotation` or `animateToRotation`, same for zoom. 

There should probably be one of these for tilt as well but I don't see it, am I overlooking it?

File: ios/src/TGMapViewController.mm (hunk 3 above)
Comment (msmollin) [inline ios/src/TGMapViewController.mm]: We did. Fixed.

File: ios/src/TGMapViewController.h (hunk 2 above)
Comment (msmollin) [inline ios/src/TGMapViewController.h]: I like `animateToPosition;` better, agreed. Updated.

File: ios/src/TGMapViewController.h (hunk 4 above)
Comment (msmollin) [inline ios/src/TGMapViewController.h]: Agreed, and updated.

As for cameraTilt, yep I missed exposing the eased variants. Good catch. Added.

File: ios/src/TGMapViewController.h (hunk 1 above)
Comment (msmollin) [inline ios/src/TGMapViewController.h]: Fair enough. Changed it to `tilt`.

**Comment by msmollin** [comment]
@blair1618 update with changes.

**Comment by blair1618** [comment]
Tweaked some naming; I'm ready to merge this now as soon as builds pass.

**Pull request merged**